applications/nrf5340_audio/src/modules/hw_codec.c

```diff
@@ -316,3 +319,96 @@ int hw_codec_init(void)
 
 	return 0;
 }
+
+static int cmd_input(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	uint8_t idx;
+
+	enum hw_codec_input {
+		LINE_IN,
+		PDM_MIC,
+		NUM_INPUT,
+	};
+
+	if (argc != 2) {
+		shell_error(shell, "Wrong number of arguments provided %d", argc);
+		return -EINVAL;
+	}
+
+	if (!isdigit((int)argv[1][0])) {
```

Comment:
Have you tried to input like a letter or something?
In other words, will this kick in even if you typecast the argument to int?

Reply:
DESCRIPTION
<<isdigit>> is a macro which classifies singlebyte charset values by table
lookup.  It is a predicate returning non-zero for decimal digits, and 0 for
other characters.

But I guess I can test it to be extra sure

Reply:
So this is tested OK?

Reply:
Yes